### PR TITLE
HADOOP-19238. Fix create-release script for arm64 based MacOS

### PR DESCRIPTION
**Pull request by mukund-thakur**
### Description of PR
Add proper checks for arm64 based machine.

### How was this patch tested?
Tested on MacOS

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  0s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m  1s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  35m  5s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 34s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  32m 29s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 38s |  |  The patch does not generate ASF License warnings.  |
|  |   |  86m 29s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6962/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6962 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs |
| uname | Linux 911aeda96bcb 5.15.0-106-generic #116-Ubuntu SMP Wed Apr 17 09:17:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / d1a64db91f2f1d3e4cee50be80a96c7ce0790aa1 |
| Max. process+thread count | 730 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6962/1/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/bin/create-release [hunk 1]
```diff
@@ -523,7 +523,6 @@ function dockermode
     echo "USER ${user_name}"
     printf "\n\n"
   ) | docker build -t "${imgname}" -f - "${BASEDIR}"/dev-support/docker/
-
```
Comment (pan3793) [inline dev-support/bin/create-release]: nit: unnecessary change

**Review by Hexiaoqiao — COMMENTED** [review]
+1 once fix the unnecessary change @pan3793 mentioned above.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   7m 28s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |   0m 40s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |   0m 40s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m 22s |  |  ASF License check generated no output?  |
|  |   |  11m 21s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6962/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6962 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs |
| uname | Linux 39129563af9a 5.15.0-106-generic #116-Ubuntu SMP Wed Apr 17 09:17:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / bd3eca591ed01fe61b7fec187f32bbbc1ce9d813 |
| Max. process+thread count | 46 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6962/2/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by steveloughran — APPROVED** [review]
LGTM
+1

**Comment by steveloughran** [comment]
merged to trunk and testing; you can backport
